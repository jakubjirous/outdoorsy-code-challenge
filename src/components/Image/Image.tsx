import React, { useCallback, useState } from "react";
import styled from "styled-components";
import { pxToRem } from "../../utils/pxToRem";

export const IMAGE_WIDTH = 180;
export const IMAGE_HEIGHT = 135;

const S = {
  Wrapper: styled.div`
    position: relative;
  `,
  Figure: styled.figure``,
  Image: styled.img`
    border-radius: var(--br);
    width: ${pxToRem(IMAGE_WIDTH)};
    height: ${pxToRem(IMAGE_HEIGHT)};
    object-fit: cover;
    object-position: center;
  `,
};

interface Props {
  src: string;
  alt: string;
}

// TODO: unit test (Jakub Jirous 2022-05-10 16:17:49)
// TODO: skeleton (Jakub Jirous 2022-05-10 16:17:51)
function Image({ src, alt }: Props) {
  const [loading, setLoading] = useState<boolean>(true);

  const onLoad = useCallback(() => {
    setLoading(false);
  }, []);

  return (
    <S.Wrapper>
      {loading && <>Loading ...</>}

      <S.Figure>
        <S.Image src={src} alt={alt} aria-label={alt} onLoad={onLoad} onError={onLoad} />
      </S.Figure>
    </S.Wrapper>
  );
}

export default Image;
