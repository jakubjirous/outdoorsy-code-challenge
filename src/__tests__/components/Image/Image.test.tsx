import { render } from "@testing-library/react";
import { ComponentProps } from "react";
import Image from "../../../components/Image/Image";

const initialProps: ComponentProps<typeof Image> = {
  src: "https://res.cloudinary.com/outdoorsy/image/upload/v1578298549/p/rentals/131280/images/ae9pnscbky1ozyhrbbtf.jpg",
  alt: "Avi",
};

const renderContainer = (props: ComponentProps<typeof Image>) => {
  return render(<Image {...props} />);
};

describe("<Image/>", () => {
  test("should render correctly", () => {
    const { baseElement } = renderContainer(initialProps);

    expect(baseElement).toMatchSnapshot();
  });
});
