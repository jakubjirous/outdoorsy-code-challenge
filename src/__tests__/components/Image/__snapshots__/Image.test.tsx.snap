// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image/> should render correctly 1`] = `
<body>
  <div>
    <div
      class="sc-bczRLJ WCHmy"
    >
      Loading ...
      <figure
        class="sc-gsnTZi"
      >
        <img
          alt="Avi"
          aria-label="Avi"
          class="sc-dkzDqf jAzbYA"
          src="https://res.cloudinary.com/outdoorsy/image/upload/v1578298549/p/rentals/131280/images/ae9pnscbky1ozyhrbbtf.jpg"
        />
      </figure>
    </div>
  </div>
</body>
`;
